packages/hdwallet-provider/src/index.ts

```diff
@@ -143,7 +135,8 @@ class HDWalletProvider {
             cb(null, tmpWallets[address].getPrivateKey().toString("hex"));
           }
         },
-        signTransaction(txParams: any, cb: any) {
+        async signTransaction(txParams: any, cb: any) {
```

Comment:
@davidmurdoch you said this was cool?

Reply:
Yup. This should be fine, as Web3 ProviderEngine is 100% callback style and ignores the returned Promise, and I can't foresee them updating the library to support Promises.